contracts/Factory.sol

```diff
@@ -27,27 +38,143 @@ contract Factory {
             // It is not generally possible to determine what is a legitmate
             // swarm hash which just happens to contain a jump destination and
             // what is malicious code.
-            if (ins == 0xf0) {return 1;} // CREATE
-            if (ins == 0xf1) {return 2;} // CALL
-            if (ins == 0xf2) {return 3;} // CALLCODE
-            if (ins == 0xf4) {return 4;} // DELEGATECALL
-            if (ins == 0xff) {return 5;} // SUICIDE
+
+            // This presented as a whitelist in case any new state-changing
+            // opcodes are added (CREATE2 being a good example)
```

Comment:
Is there any metrics we could go by to see how this increases the gas cost?

Reply:
I'm assuming we remove this for prod if the costs are high.

Reply:
We don't have any as yet, though that's obviously something we need to do for many reasons. Given the example of CREATE2 I think whitelisting is completely necessary. There is some scope for optimisation to use ranges rather than testing for each. For some reason solidity doesn't have a switch statement.

If we really want to make this cheaper we could do it in an assembly switch statement which will be almost as low as one instruction per opcode.

Reply:
This whitelist wouldn't work currently since you are only checking the opcodes that are known. I would do a blacklist instead that simply doesn't allow any unknown opcodes.